functional/keylime_tenant-ima-signature-sanity/test.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
+
+. /usr/share/beakerlib/beakerlib.sh || exit 1
+
+# This test requires HW TMP
+
+AGENT_ID="d432fbb3-d2f1-4a97-9ef7-75bd81c00000"
+
+rlJournalStart
+
+    rlPhaseStartSetup "Do the keylime setup"
+        rlRun 'rlImport "./test-helpers"' || rlDie "cannot import keylime-tests/test-helpers library"
+        rlAssertRpm keylime
+        limeBackupConfig
+        # update /etc/keylime.conf
+        rlRun "limeUpdateConf tenant require_ek_cert False"
+        # if TPM emulator is present
+        if limeTPMEmulated; then
+            # start tpm emulator
+            rlRun "limeStartTPMEmulator"
+            rlRun "limeWaitForTPMEmulator"
+            # make sure tpm2-abrmd is running
+            rlServiceStart tpm2-abrmd
+            sleep 5
+            # start ima emulator
+            rlRun "limeInstallIMAConfig"
+            rlRun "limeStartIMAEmulator"
+        fi
+        sleep 5
+        # start keylime_verifier
+        rlRun "limeStartVerifier"
+        rlRun "limeWaitForVerifier"
+        rlRun "limeStartRegistrar"
+        rlRun "limeWaitForRegistrar"
+        rlRun "limeStartAgent"
+        rlRun "limeWaitForAgentRegistration ${AGENT_ID}"
+        # create allowlist and excludelist
+        limeCreateTestLists
+        # start simple http server to serve files
+        rlRun "python3 -m http.server 8000 &"
+        HTTP_PID=$!
+        #add GPG_TTY variables
+        rlRun "GPG_TTY=$(tty)"
+        rlRun "export GPG_TTY"
+        # generate GPG key
+        rlRun "gpgconf --kill gpg-agent"
+        rlRun "export GNUPGHOME=${TmpDir}"
+        rlRun "cat >gpg.script <<EOF
+%echo Generating a basic OpenPGP key
+Key-Type: RSA
+Key-Length: 4096
+Subkey-Type: RSA
+Subkey-Length: 4096
+Name-Real: Joe Tester
+Name-Comment: with no passphrase
+Name-Email: joe@foo.bar
+Expire-Date: 0
+# Do a commit here, so that we can later print 'done' :-)
+%commit
+%echo done
+EOF"    
+        #generate gpg keys which will be genuine
+        rlRun "gpg --batch --pinentry-mode=loopback --passphrase '' --generate-key gpg.script"
+        rlRun "gpg --armor -o gpg-key-genuine.pub --export joe@foo.bar"
+        # sign our IMA key file
+        rlRun "gpg --detach-sign -o signature-gpg-genuine.sig ${limeIMAPublicKey}"
+        #generate fake gpg key
+        rlRun "gpg --batch --pinentry-mode=loopback --passphrase '' --generate-key gpg.script"
```

Comment:
I am not sure if this is correct. You are generating another key that cannot be easily distinguished from the previous one since the config file is identical. Later, when exporting the key you can't be sure which key you are in fact exporting.
Most importantly, you should not need another signature and key. For testing negative scenario you need just a different IMA key.